### PR TITLE
keep BOOT_ACK and PARTITION_ACCESS values 

### PR DESCRIPTION
**Pull request by s-michaelsen**
when writing number of active boot partition to eMMCs EXT_CSD_PART_CONFIG register

### CODE REVIEW

**Comment by ejoerns** [comment]
@s-michaelsen please add your Signed-of-by line here (https://rauc.readthedocs.io/en/latest/contributing.html#developer-s-certificate-of-origin)

**Comment by jluebbe** [comment]
For the PARTITION_ACCESS, this should not be needed, as the kernel is responsible for maintaining it (https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a0e9576634f0bcfa5c4d8efdcf8c9b28298d7570). It would be vulnerable to races between userspace an kernel anyway. Did you see problems with BOOT_ACK only, or also with PARTITION_ACCESS?
Regarding the BOOT_ACK value, I agree. A comment in the code explaining this would be useful to avoid problems in the future.

**Comment by s-michaelsen** [comment]
I saw only problems with BOOT_ACK flag. Only added PARTITION_ACCESS as well after reading register spec. I'll remove the according part and change the comment.

**Comment by jluebbe** [comment]
I've squashed the commits together and reworded the comment a bit.

File: src/emmc.c [hunk 1]
```diff
@@ -99,11 +100,26 @@ gboolean r_emmc_write_bootpart(const gchar *device, gint bootpart_active, GError
...
+				EXT_CSD_PART_CONFIG, device);
+		goto out;
+	}
+
+	/* Keep BOOT_ACK value as it is. Resetting this bit might prevent
+	 * proper boot process on some platforms since in some SoCs, ROM boot
+	 * loaders need this flag to be set.
+	 * The PARTITION_ACCESS value is handled inside the kernel so we do not
+	 * need to take care of it here. */
+	value = extcsd[EXT_CSD_PART_CONFIG] & ~(1 << 6);
```
Comment (ejoerns) [inline src/emmc.c]: This clears `BOOT_ACK` which is the opposite of what we intended to achieve.

Shouldn't that be:
``` suggestion
	value = extcsd[EXT_CSD_PART_CONFIG] & (1 << 6);
```

**Comment by ejoerns** [comment]
I've squashed the commits and rebased onto current master.

**Pull request merged**